kedro/framework/project/__init__.py

```diff
@@ -254,3 +256,53 @@ def validate_settings():
     More info on the dynaconf issue: https://github.com/rochacbruno/dynaconf/issues/460
     """
     importlib.import_module(f"{PACKAGE_NAME}.settings")
+
+
+def find_pipelines() -> Dict[str, Pipeline]:
+    """Automatically find modular pipelines having a ``create_pipeline``
+    function. By default, projects created using Kedro 0.18.3 and higher
+    call this function to autoregister pipelines upon creation/addition.
+
+    Returns:
+        A generated mapping from pipeline names to ``Pipeline`` objects.
+
+    Warns:
+        UserWarning: When a module does not expose a ``create_pipeline``
+            function, the ``create_pipeline`` function does not return a
+            ``Pipeline`` object, or if the module import fails up front.
+    """
+    pipelines_dict = {"__default__": pipeline([])}
+    for pipeline_name in importlib.resources.contents(f"{PACKAGE_NAME}.pipelines"):
+        try:
+            pipeline_module = importlib.import_module(
+                f"{PACKAGE_NAME}.pipelines.{pipeline_name}"
+            )
+        except ModuleNotFoundError:
+            continue
```

Comment:
When would this be raised and why don't we want to emit a warning in this case?

Reply:
I killed it, with the move to `iterdir`. :)